File: pkgs/types-py/module/template.py
from .arg import Arg
from typing import Optional
from .location import Location
from genotype import Model


class Template(Model):
    id: str
    name: str
    description: str
    args: list[Arg]
    lang: Optional[str] = None
    content: str
    location: Location


class TemplateCollection(Model):
    name: str
    description: str
    templates: list[Template]
    location: Location

File: pkgs/types-rs/src/template.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct Template {
    pub id: String,
    pub name: String,
    pub description: String,
    pub args: Vec<super::arg::Arg>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub lang: Option<String>,
    pub content: String,
    pub location: super::location::Location,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct TemplateCollection {
    pub name: String,
    pub description: String,
    pub templates: Vec<Template>,
    pub location: super::location::Location,
}
